src/wallet/rpcwallet.cpp

```diff
@@ -734,7 +735,11 @@ static UniValue getbalance(const JSONRPCRequest& request)
         filter = filter | ISMINE_WATCH_ONLY;
     }
 
-    return ValueFromAmount(pwallet->GetBalance(filter, min_depth));
+    bool avoid_reuse = request.params[3].isNull()
+        ? pwallet->IsWalletFlagSet(WALLET_FLAG_AVOIDREUSE)
+        : request.params[3].get_bool();
```

Comment:
Above help text says `(only applicable if the \"avoid reuse\" wallet flag is enabled)`, so why do you have this case? Shouldn't we just error immediately if it is set without `pwallet->IsWalletFlagSet(WALLET_FLAG_AVOIDREUSE)`? Same with the other RPCs

Reply:
Code now throws an error if user does `avoidreuse=true` for a wallet which does not have `WALLET_FLAG_AVOID_REUSE` enabled.